### PR TITLE
Switch Script

### PR DESCRIPTION
**Pull request by elie0174**
This is the code for the switches

### CODE REVIEW

**Comment by Thanh2311** [comment]
can u update requirements.txt

**Review by Thanh2311 — APPROVED** [review]

File: project/scripts/SwitchScript.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import time
+from kasa import SmartPlug
+from kasa import Discover
+# to use pip install python-kasa
+#call this function fo find all the ips and switch names in found_devices.items()
+async def findDevices():
+    found_devices = asyncio.run(Discover.discover())
+    return found_devices
+
+#call this function with the the state you want the switch in and its ip to turn off/on the switch
```
Comment (erick576) [inline project/scripts/SwitchScript.py:11]:
```suggestion
#  This function takes in the IP address and state of a switch to then turn it on/off
```

File: project/scripts/SwitchScript.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import asyncio
+import time
+from kasa import SmartPlug
+from kasa import Discover
+# to use pip install python-kasa
```
Comment (erick576) [inline project/scripts/SwitchScript.py:5]:
```suggestion

```
No need as they will install through Requirements.txt

File: project/scripts/SwitchScript.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+#call this function fo find all the ips and switch names in found_devices.items()
```
Comment (erick576) [inline project/scripts/SwitchScript.py:6]:
```suggestion
# This function discovers all the IP's and switch names
```

File: project/scripts/SwitchScript.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+async def findDevices():
+    found_devices = asyncio.run(Discover.discover())
+    return found_devices
+
+#call this function with the the state you want the switch in and its ip to turn off/on the switch
+async def ConsolSwitch(state,IP):
```
Comment (erick576) [inline project/scripts/SwitchScript.py:12]:
```suggestion
async def ConsolSwitch(state, IP):
```

**Comment by erick576** [comment]
Looks good, just have some small cleanup suggestions on the code

**Review by digengill — APPROVED** [review]

**Comment by erick576** [comment]
Looks good @elie0174 can you please [squash](https://www.youtube.com/watch?v=V5KrD7CmO4o) the commits before merging to the master branch? Ideally we want to only have extra commits if they are reasonable to be separated. All spelling check or addressing feedback should just be squashed into the initial commit, thanks

**Review by ChandlerCabrera — APPROVED** [review]

**Review by AlexCHLo — APPROVED** [review]
Looks good.

**Pull request merged**